towerlib/towerlib.py

```diff
@@ -1997,24 +2011,579 @@ def schedules(self):
+            return response.json()
+
+    def change_project_of_job_template(self, job_template, project_id):
+        """A job template in the ansible tower has to have a project from the list of projects.
+
+        This function helps to change the project name of a job template in ansible tower for a given job template.
+
+        Args:
+            job_template: given job template object
+            project_id: given project's id
+
+        Returns:
+            Object: The updated job_template object
+
+        """
+
+        project_obj = self.get_project_by_id(project_id)
+        project_playbooks = self.get_playbooks_by_branch_name(project_obj.scm_url, project_obj.scm_branch)
+        new_playbook = job_template.playbook
+        if new_playbook not in project_playbooks:
+            self._logger.warning("Job template's playbook '{}' is not found in the given project's playbook"
+                                 "list. Now changing the playbook to '{}'. Please select a playbook manually."
+                                 .format(new_playbook, project_playbooks[0]))
+            new_playbook = project_playbooks[0]
+
+        new_data = {
+            "id": job_template.id,
+            "project": project_obj.id,
+            "playbook": new_playbook,
+            "name": job_template.name,
+            "description": job_template.description
+        }
+        response = self.change_job_template_data(new_data)
+        if response is not None:
+            self._logger.info("The project for the job template '{}' is changed from '{}' to '{}'".
+                              format(job_template.name, job_template.project.name, project_obj.name))
+        else:
+            self._logger.error("Error updating the job template's project.")
+        return response
+
+    def change_project_of_job_template_to_prod_branch_project(self, job_template, scm_url):
+        """A job template in the ansible tower has to have a project from the list of projects.
+
+        This function helps to change the project name of a job template in ansible tower for a given job template.
+
+        Args:
+            job_template: given job template object
+            scm_url: the scm url of the projects.
+
+        Returns:
+            Object: The updated job_template object
+
+        """
+        prod_branch_project_id = self.get_production_branch_project_id(scm_url)
+        response = self.change_project_of_job_template(job_template, prod_branch_project_id)
+        return response
+
+    def change_job_type(self, given_labels, new_job_type):
+        """Change the job template type in the given Ansible Tower, which means whether a job should run in 'Run' or
+        'Check' mode.
+
+        Args:
+            given_labels: ist of labels which need to match
+            new_job_type: the updated job type.
+
+        Returns:
+
+        """
+        count = 0
+        job_templates = list(self.job_templates)
```

Comment:
Too broad scope. Also blocking and not required to build a list, you can just iterate over the attribute since it is a generator.

Reply:
Implemented iteration over generator.